api/ocm/tools/signing/digest.go

```diff
@@ -78,6 +67,28 @@ func VerifyResourceDigest(cv ocm.ComponentVersionAccess, i int, bacc ocm.DataAcc
 	return true, nil
 }
 
+// VerifyResourceDigest does essentially the same as VerifyResourceDigestForResource but takes an optional VerifiedStore
+// to check the digest against a verified version of the component version.
+func VerifyResourceDigest(cv ocm.ComponentVersionAccess, i int, bacc ocm.DataAccess, ostore ...VerifiedStore) (bool, error) {
+	cd := cv.GetDescriptor()
+	raw := &cd.Resources[i]
+
+	store := general.Optional(ostore...)
+	if store != nil {
+		vcd := store.Get(cv)
+		if vcd == nil {
+			return false, fmt.Errorf("component version %s not verified", common.VersionedElementKey(cv))
+		}
+		if !vcd.Resources[i].Digest.Equal(raw.Digest) {
+			return false, fmt.Errorf("component version %s corrupted", common.VersionedElementKey(cv))
+		}
+
+		return true, nil
```

Comment:
if the store is ok, but the cv not found, the verification should happen as if no store was provided

```suggestion
		if vcd != nil {
			return vcd.Resources[i].Digest.Equal(raw.Digest), nil
		}
```